Idno/Common/Page.php

```diff
@@ -847,14 +851,40 @@ function adminGatekeeper()
             }
         }
 
+        /**
+         * Sets the entity on the page to the specified object
+         * @param object $entity
+         */
+        function setEntity($entity) {
+            $this->entity = $entity;
+        }
+
+        /**
+         * Returns the entity associated with this page, if it exists
+         * @param $entity
+         * @return object|null
+         */
+        function getEntity($entity) {
+            return $this->entity;
+        }
+
+        /**
+         * Removes any entity associated with this page
+         */
+        function removeEntity() {
+            $this->entity = null;
+        }
+
         /**
          * Is this page a permalink for an object? This should be set to 'true'
-         * if it is.
+         * if it is. Optionally, we can also associate the page with the object here.
          * @param bool $status Is this a permalink? Defaults to 'true'
+         * @param object $entity Optionally, an entity this page is associated with
          */
-        function setPermalink($status = true)
+        function setPermalink($status = true, $entity = null)
```

Comment:
Consider using type hinting (as per style guidelines)

(this is relatively new guideline, so you may not be aware!)

Reply:
Updated